packages/angular/projects/ui-angular/src/lib/primitives-poc/view/view-component/view.component.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  HostBinding,
+  Input,
+  OnChanges,
+  OnInit,
+  Renderer2,
+  SimpleChanges,
+  ViewEncapsulation,
+} from '@angular/core';
+
+/** Config file to get all the configuration related the strings and values */
+import { ComponentSettings } from './ComponentSetting';
+@Component({
+  selector: ComponentSettings.COMPONENTNAME,
+  template: '<ng-content></ng-content>',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AmplifyViewComponent implements OnInit, OnChanges {
+  /** Input Prop for style changes  */
+  @Input() type: 'submit' | 'button' = 'button';
+  @Input() fullWidth: boolean | string = false;
+  @Input() isDisabled: boolean | string = false;
+  @Input() size: 'small' | 'medium' | 'large' = 'medium';
+  @Input() variation: 'primary' | 'default' | 'link' = 'default';
+
+  @HostBinding('type') typeAttr: string;
+  @HostBinding('attr.data-fullwidth') fullWidthAttr: boolean | string;
+  @HostBinding('attr.data-size') sizeAttr: string;
+  @HostBinding('attr.data-variation') variationAttr: string;
+  @HostBinding('style.font-weight') fontWeightAttr: string;
```

Comment:
Same over here -- these are button props and not view props.

Reply:
Done.